### PR TITLE
Add info about v2 VPI not support in CSI

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:
Add no support in CSI cinder on cinder v2 API 
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: docs/provider-configuration.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ a future release. As of the "Queens" release, OpenStack no longer exposes the
 Identity v2 API.
 
 ‡ Block Storage v3 API support was added in Kubernetes 1.9.
+  For CSI cinder enablement, only cinder v3 API is supported currently.
```
Comment (adisky) [inline docs/provider-configuration.md]: NOTE: Cinder CSI Supports Block Storage V3 API only

File: docs/provider-configuration.md [hunk 2]
```diff
@@ -44,6 +44,7 @@ a future release. As of the "Queens" release, OpenStack no longer exposes the
 Identity v2 API.
 
 ‡ Block Storage v3 API support was added in Kubernetes 1.9.
+  *NOTE*: Cinder CSI Supports Block Storage V3 API only.
```
Comment (ramineni) [inline docs/provider-configuration.md]: Isn't it better to add this info in cinder csi plugin doc https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-cinder-csi-plugin.md ?

And also formatting seems wrong (it is coming in same line as L46),  I guess there should be blank line above i guess .

Comment (adisky) [reply docs/provider-configuration.md]: we can add this in CSI docs also, but many people land on this page for provider configuration and assumes the same configuration for intree/CSI driver.

Comment (ramineni) [reply docs/provider-configuration.md]: Looks like we need to do some refactoring here,  for now this PR should be ok once formatting is fixed. Thanks

Comment (jichenjc) [reply docs/provider-configuration.md]: @ramineni updated just now

**Comment by ramineni** [comment]
Thanks
/lgtm

**Pull request merged**